pkg/network/protocols/core/protocol.go

```diff
@@ -200,10 +202,37 @@ func (p *Protocol) onAttestations(commitmentBytes []byte, attestationsBytes []by
 		return
 	}
 
-	var attestations []*iotago.Attestation
-	// TODO: there could be multiple versions of attestations in the same packet
-	if _, err := lo.PanicOnErr(p.apiProvider.APIForVersion(iotago.Version(commitmentBytes[0]))).Decode(attestationsBytes, &attestations, serix.WithValidation()); err != nil {
-		p.Events.Error.Trigger(ierrors.Wrap(err, "failed to deserialize attestations"), id)
+	if len(attestationsBytes) < 4 {
+		p.Events.Error.Trigger(ierrors.Errorf("failed to deserialize attestations, invalid attestation count"), id)
+
+		return
+	}
+
+	attestationCount := binary.LittleEndian.Uint32(attestationsBytes[0:4])
```

Comment:
Checking for the constant, using `binary.LittleEndian` package and having a manual `readOffset` seems to be a bit cumbersome. I think this would be much nicer with the `stream` package, here specifically `ReadCollection`